academy/xl-run-prod/3-genesis.md

```diff
@@ -0,0 +1,222 @@
+
+### Genesis accounts
+
+Genesis accounts are accounts that exist in the genesis. They can be there because of a pre-sale of tokens, because of simple allotments or for any other reason. You need to include them in the genesis. The command for that is `add-genesis-account`. For each genesis account you need to collect their address. If these are third-parties, make sure in more ways than one that you get the right values, whether by email or other means.
+
+Genesis accounts, like any account, have addresses. Your addresses have a prefix. Suppose you chose `cavedweller` for your blockchain, instead of the Cosmos Hub default of `cosmos`.
+
+Here you can decide to allocate new tokens to your genesis accounts. There is no limit to the number of genesis accounts and the number of extra tokens in the genesis. If you introduce another token named `nflint`, then if Alice has the address `cavedweller1nw793j9xvdzl2uc9ly8fas5tcfwfetercpdfqq` you could make her a genesis account with:
+
+```sh
+$ ./myprojectd add-genesis-account cavedweller1nw793j9xvdzl2uc9ly8fas5tcfwfetercpdfqq 5000000000stone 2000000000nflint
+```
+
+This credits her with 5 STONE and 2 FLINT. It has also given her an `account_number` in the genesis.
+
+You can do the same for all your genesis accounts, which includes your validators.
+
+Advanced topics include [vesting](https://docs.cosmos.network/main/modules/auth/05_vesting.html) on these genesis accounts. For information on how to configure them do:
+
+```sh
+$ ./myprojectd add-genesis-account --help
+```
+
+When you have defined the genesis accounts, it is time to define the genesis validators.
+
+### Validator stakes
+
+The genesis needs to define the starting validators. That's because, at least, a validator needs to propose the first block that comes after the genesis. But if no validators have been agreed on by consensus, then the first block cannot be produced.
+
+Assume that you have a team of validators that are your starting validators. They are also genesis accounts that you defined previously, so they _have_ tokens. Now you have to:
+
+1. Credit each genesis validator the agreed staked token amounts.
+2. Collect signed transactions from them that identify them as validators.
+
+Each validator needs to generate a transaction locally and send it to you, the party that collects them into the genesis.
+
+To each validator you send:
+
+* The `account_number` that was given to them when calling `add-genesis-account`.
+* A confirmation of the amount of tokens that you have credited them.
+
+Each validator operator then has to run a `gentx` command. This command is not to be run on the server. Instead it is run on the computer that hosts the _cold_ validator operator app key. Using the keyring of your choice. Collect the consensus public key from Tendermint KMS, for instance `{"@type":"/cosmos.crypto.ed25519.PubKey","key":"byefX/uKpgTsyrcAZKrmYYoFiXG0tmTOOaJFziO3D+E="}`.
```

Comment:
Sentence fragment: "Using the keyring of your choice." Not clear whether this should be expanded, or become part of the sentence before, or part of the sentence after. Possible options:

1: ...it is run on the computer that hosts the _cold_ validator operator app key, using the keyring of your choice.

2. Using the keyring of your choice, collect the consensus public key from Tendermint KMS...

Reply:
Oops. Option 1.